### PR TITLE
Add an expert mode splash screen before the parameter field of the transaction operation

### LINKED ISSUE #175: Add an expert mode splash screen before the `parameter` field of the transaction operation
According to https://github.com/trilitech/ledger-app-tezos-wallet/pull/173#discussion_r1431237385, the 'parameter' field of the transaction operation must be preceded by an expert mode splash screen.

### PR DESCRIPTION
**Pull request by spalmer25**
fixes #175

### CODE REVIEW

File: app/src/parser/operation_state.h [hunk 1]
```diff
@@ -87,10 +91,19 @@ typedef enum {
...
 } tz_operation_field_kind;
 
-typedef struct {
+typedef struct tz_operation_field_descriptor {
     const char             *name;
     tz_operation_field_kind kind : 5;
-    uint8_t required : 1, skip : 1, display_none : 1, complex : 1;
+    union {
+        struct {
+            const struct tz_operation_field_descriptor *field;
```
Comment (ajinkyaraj-23) [inline app/src/parser/operation_state.h:99]: Field_option and step_option has duplicate syntax. Its better to define the struct seperately and use it inside tz_operation_field_descriptor and  tz_operation_parser_frame

Comment (ajinkyaraj-23) [reply app/src/parser/operation_state.h:99]: Not a critical change right now.

**Review by ajinkyaraj-23 — APPROVED** [review]
overall LGTM. 
Step by step commits are great to understand incremental changes done.

File: app/src/parser/operation_parser.c [hunk 2]
```diff
@@ -47,179 +49,151 @@ const char *const tz_operation_parser_step_name[] = {"MAGIC",
...
-    {"Storage",       TZ_OPERATION_FIELD_EXPR,   true,  false, false, true },
-    {NULL,            0,                         0,     0,     0,     0    }
-};
+TZ_OPERATION_FIELDS(transaction_fields,
+    TZ_OPERATION_MANAGER_OPERATION_FIELDS,
+    TZ_OPERATION_FIELD("Amount",      TZ_OPERATION_FIELD_AMOUNT),
+    TZ_OPERATION_FIELD("Destination", TZ_OPERATION_FIELD_DESTINATION),
+    TZ_OPERATION_OPTION_FIELD("_Parameters",
+        TZ_OPERATION_TUPLE_FIELD("_Parameters",
+            TZ_OPERATION_FIELD("Entrypoint", TZ_OPERATION_FIELD_SMART_ENTRYPOINT, .complex=true),
```
Comment (emturner) [inline app/src/parser/operation_parser.c:118]: I think `Entrypoint` can be made `complex=false` 😄

Comment (spalmer25) [reply app/src/parser/operation_parser.c:118]: I agree, let's do it in another PR: #194

File: app/src/parser/operation_state.h (hunk 1 above)
Comment (spalmer25) [reply app/src/parser/operation_state.h]: https://github.com/trilitech/ledger-app-tezos-wallet/compare/21567e5d6dad4dfb8e429d8480ef2cd3bc9d803d..64bd1070d734439b830c8ce8de7868554b35f82b

**Pull request merged**